tests/boundary_composition-passive_sphere.prm

```diff
@@ -0,0 +1,112 @@
+#########################################################
+# This is a variation of the platelike-boundary.prm file.
+# All settings are exactly the same with the exception
+# of the ones that pertain to compositional fields.
+#
+# See the manual for more information about this setup.
+
+
+set Dimension                              = 2
+set Start time                             = 0
+set End time                               = 20
+set Use years in output instead of seconds = false
+set Output directory                       = output-boundary_composition-passive_sphere
+
+
+
+subsection Geometry model
+  set Model name = spherical shell
+
+  subsection Spherical shell
+    set Inner radius  = .45
+    set Outer radius  = 1
+    set Opening angle = 90
+  end
+end
+
+
+subsection Model settings
+  set Zero velocity boundary indicators       = inner
+  set Tangential velocity boundary indicators = outer
+  set Prescribed velocity boundary indicators =
+  set Fixed temperature boundary indicators   = inner, outer
+  set Fixed composition boundary indicators   = inner, outer
+end
+
+
+subsection Boundary temperature model
+  set Model name = spherical constant
+  subsection Spherical constant
+    set Inner temperature = 1
+    set Outer temperature = 0
+  end
+end
+
+
+subsection Gravity model
+  set Model name = radial constant
+
+  subsection Vertical
+    set Magnitude = 9.81
+  end
+end
+
+
+subsection Initial temperature model
+  set Model name = function
+
+  subsection Function
+    set Coordinate system   = spherical
+    set Variable names      = r,phi
+    set Function expression = (1-r)
+  end
+end
+
+
+subsection Material model
+  set Model name = simple
+
+  subsection Simple model
+    set Thermal conductivity          = 1e-6
+    set Thermal expansion coefficient = 1e-4
+    set Viscosity                     = 1
+  end
+end
+
+
+subsection Mesh refinement
+  set Initial adaptive refinement        = 0
+  set Initial global refinement          = 5
```

Comment:
that's a lot of refinement, and consequently a lot of data that goes with the test. can you reduce this number? To verify the correctness of your class, you don't actually need a very fine mesh :-)